torch/lib/ATen/templates/TypeDerived.cpp

```diff
@@ -33,7 +34,10 @@ std::unique_ptr<Storage> ${Type}::storageFromBlob(void * data, int64_t size) {
     return std::unique_ptr<Storage>(
       new ${Storage}(context,data,size));
 }
-Tensor ${Type}::unsafeTensorFromTH(void * th_pointer) {
+// Calls retain on th_pointer
+Tensor ${Type}::unsafeTensorFromTH(void * th_pointer, bool retain) {
+  if (retain)
+    ${THTensor}_retain(${state,} (${THTensor}*) th_pointer);
   return Tensor(new ${Tensor}(context,(${THTensor}*)(th_pointer)));
```

Comment:
This is where a memory leak is happening. Should be `return Tensor(new ${Tensor}(context,(${THTensor}*)(th_pointer)),false);`. The `false` ensures Tensor doesn't bump the refcount of the thing that was just allocated. This is fixed in aten/master and pytorch master, I am just pointing it out here so that when you merge them in you know what to look for.